Assets/Scripts/Character.cs

```diff
@@ -1,13 +1,19 @@
 using System;
+using System.Collections;
 using System.Collections.Generic;
+using UnityEngine;
 
 public class Character
 {
-    List<Critter> _critters;
-    Critter _currentCritter;
-
-    public virtual void DoTurn(ICharacterBattleManager bm)
+    public string Name { get; private set; }
+    
+    protected List<Critter> _critters;
+    protected Critter _currentCritter;
+    
+    public virtual void Act(ICharacterBattleManager bm)
     {
-        throw new NotImplementedException();
+        OnAct();
```

Comment:
If you want this to be an event that is triggered regardless of the override by different implementations of character, it might be better to have something like this:
* a public virtual OnBeforeAct() event.
* a public virtual DoAct() function.
* a public *non-virtual* act function that calls the OnBeforeAct() event followed by the DoAct() function.

This way implementers can override the base class DoAct but the event is still called for all of them without them having to do it manually by calling the base class implementation